frontend/document/document.lua

```diff
@@ -39,6 +39,8 @@ function Document:new(from_o)
     self.__index = self
     if o._init then o:_init() end
     if o.init then o:init() end
+    -- Make sure the checksum has been generated before any document saving operations.
+    o:fastDigest()
```

Comment:
@houqp If he implements it the way you suggest then is there a way to NOT call all these handlers if the user doesn't want to use kosync at all? That would be really useful.